src/main/java/org/lisoft/lsml/view_fx/controls/EquippedItemCell.java

```diff
@@ -318,22 +331,60 @@ private boolean onRemoveItem() {
         return safeCommand(this, stack, new CmdRemoveItem(msgd, loadout, component, getItem()), msgd);
     }
 
-    private void setupEngineRatingDropDown(boolean aPgiMode) {
+    private <T> boolean addIfEngineExists(final ObservableList<T> aList, final T aValue, final Integer aRating, final EngineType aType, final Faction aFaction) {
+        try {
+            if (aRating != null && aType != null && aFaction != null) {
+                ItemDB.getEngine(aRating, aType, aFaction);
+            }
+            aList.add(aValue);
+            return true;
+        } catch (NoSuchItemException ex) {
+            return false;
+        }
+    }
+
+    private void regenerateEngineRatingDropDown() {
         if (loadout.getChassis() instanceof ChassisStandard) {
             final ChassisStandard chassis = (ChassisStandard) loadout.getChassis();
+
+            final Engine selectedEngine = loadout.getEngine();
```

Comment:
If engine is null, just return without doing anything, otherwise all the work below is performed for each item slot on the mech. Doing so also makes the null checking below and in `addIfEngineExists` unnecessary.